fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt

```diff
@@ -741,4 +748,115 @@ class BookmarksTest {
             verifyFolderTitle("My Folder")
         }
     }
+
+    @Test
+    fun verifySearchBookmarksViewTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        browserScreen {
+            createBookmark(defaultWebPage.url)
+        }.openThreeDotMenu {
+        }.openBookmarks {
+            clickSearchButton()
+            verifyBookmarksSearchBar(true)
+            verifyBookmarksSearchBarPosition(true)
+            clickOutsideTheSearchBar()
+            verifyBookmarksSearchBar(false)
+        }.goBackToBrowserScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openCustomizeSubMenu {
+            clickTopToolbarToggle()
+        }
+
+        exitMenu()
+
+        browserScreen {
+        }.openThreeDotMenu {
+        }.openBookmarks {
+            clickSearchButton()
+            verifyBookmarksSearchBar(true)
+            verifyBookmarksSearchBarPosition(false)
+            dismissBookmarksSearchBarUsingBackButton()
+            verifyBookmarksSearchBar(false)
+        }
+    }
+
+    @Test
+    fun verifySearchForBookmarkedItemsTest() {
+        val firstWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        val secondWebPage = TestAssetHelper.getHTMLControlsFormAsset(mockWebServer)
+
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openBookmarks {
+            createFolder(bookmarksFolderName)
+        }
+
+        exitMenu()
+
+        browserScreen {
+            createBookmark(firstWebPage.url, bookmarksFolderName)
+            createBookmark(secondWebPage.url)
+        }.openThreeDotMenu {
+        }.openBookmarks {
+            clickSearchButton()
+            // Search for a valid term
+            searchBookmarkedItem(firstWebPage.title)
+            verifySearchedBookmarkExists(firstWebPage.url.toString(), true)
+            verifySearchedBookmarkExists(secondWebPage.url.toString(), false)
+            // Search for invalid term
+            searchBookmarkedItem("Android")
+            verifySearchedBookmarkExists(firstWebPage.url.toString(), false)
+            verifySearchedBookmarkExists(secondWebPage.url.toString(), false)
+        }
+    }
+
+    @Test
+    fun verifyVoiceSearchInBookmarksTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        browserScreen {
+            createBookmark(defaultWebPage.url)
+        }.openThreeDotMenu {
+        }.openBookmarks {
+            clickSearchButton()
+            verifyBookmarksSearchBar(true)
+            clickVoiceSearchButton()
```

Comment:
You could add here a transition to SearchRobot and reuse the startVoiceSearch method. Maybe with a little tweaking: 
`fun startVoiceSearch() {
        voiceSearchButton.click()
}`

then use TestHelper.grantPermission()
then TestHelper.assertNativeAppOpens()

you can also do this refactor in toggleVoiceSearchTest()